### PR TITLE
[Feature] ZF3 support (no BC breaks)

### PR DESCRIPTION
**Pull request by shandyDev**
Hi guys!
I updated module to support both ZF version v2 & v3 (look https://docs.zendframework.com/zend-servicemanager/migration/). 
Please review these changes and give me some feedback. If all right I propose release it with 2.1 version tag.

/cc @juriansluiman @bakura10

### CODE REVIEW

**Comment by shandyDev** [comment]
Tests was passed on both ZF versions.

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,69 @@
+# Created by .ignore support plugin (hsz.mobi)
```
Comment (bakura10) [inline .gitignore]: Not everything in here is required.

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+# workspace files are user-specific
+*.sublime-workspace
+
+# project files should be checked into the repository, unless a significant
+# proportion of contributors will probably not be using SublimeText
+# *.sublime-project
+
+# sftp configuration file
+sftp-config.json
```
Comment (bakura10) [inline .gitignore]: not needed

File: .gitignore [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+bh_unicode_properties.cache
+
+# Sublime-github package stores a github token in this file
+# https://packagecontrol.io/packages/sublime-github
+GitHub.sublime-settings
```
Comment (bakura10) [inline .gitignore]: Not needed

File: .gitignore [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+# Binary gettext files
+*.mo
+
+### JetBrains template
+# Covers JetBrains IDEs: IntelliJ, RubyMine, PhpStorm, AppCode, PyCharm, CLion, Android Studio and Webstorm
+# Reference: https://intellij-support.jetbrains.com/hc/en-us/articles/206544839
+
+# User-specific stuff:
+.idea/
```
Comment (bakura10) [inline .gitignore]: Eventually this one considering how common it is

**Comment by bakura10** [comment]
Keeping a very small .gitignore with only .idea and some composer files please. All the rest is really not reaquired and sohuld be part of your global ignore.

Otheriwse it looks like a solid update to me. @danizord have time to do another quick reading ?:)

**Comment by shandyDev** [comment]
Gitignore was cleaned. Thank you for review. )

File: composer.json [hunk 5]
```diff
@@ -30,13 +30,24 @@
         }
     ],
     "require": {
-        "php": ">=5.6",
-        "zendframework/zend-mail": "~2.0",
-        "zendframework/zend-http": "~2.0"
+        "php": "^5.6 || ^7.0",
+        "zendframework/zend-mail": "^2.6",
+        "zendframework/zend-http": "^2.5",
+        "zendframework/zend-mime": "^2.6"
```
Comment (danizord) [inline composer.json]: `"zendframework/zend-servicemanager": "^2.7 || ^3.1"` should be added here as well

**Comment by bakura10** [comment]
Tagged!

Thanks a lot :).

**Comment by shandyDev** [comment]
Thank you! You super fast guys! )

**Pull request merged**